### DISCUSSION TITLE
Invoked callbacks are just functions

### DISCUSSION THREAD

**Discussion by spankyed** [statelyai/xstate#2604] [Ideas]
Why is `Invoking callbacks` called that and not simply `Invoking functions`? And the first argument for an invoked function named `callback` instead of `send`?

> **Invoking Callbacks**
> Streams of events sent to the parent machine can be modeled via a callback handler, which is a function that takes in two arguments:

```Javascript
counting: {
  invoke: {
    id: 'incInterval',
    src: (context, event) => (callback, onReceive) => {
      // This will send the 'INC' event to the parent every second
      const id = setInterval(() => callback('INC'), 1000);

      // Perform cleanup
      return () => clearInterval(id);
    }
  },
  on: {
    INC: { actions: assign({ counter: context => context.counter + 1 }) }
  }
}
```

**Comment by ChrisShank** [comment]
> Why is Invoking callbacks called that and not simply Invoking functions? 

This usage is by definition a [callback](https://stackoverflow.com/a/7549753). 

> And the first argument for an invoked function named callback instead of send?

Ya I agree here! `callback` is ambiguous name for this argument and the docs should probably be updated. I generally call it `sendBack` since `send` is a function exported by xstate and its best to avoid potential name collisions. At the end of the day the consumer has control of what they call it what they want.
```ts
import {send} from 'xstate'
```

**Comment by spankyed** [comment]
What's being invoked is misemphasized by calling the feature `Invoking callback`. When invoking promises, what's being invoked is the body of the promise or the task that the promise is to complete, rather than the event/state transition(callback), which is an afterthought. Same for invoking machines or observables. But with invoking callbacks, the emphasis seems to be on the subsequent event/transition rather then the body of the function that executes this callback.  To be consistent, this feature would need to be called `Invoking callback handlers`. 

I think calling this feature `Invoking functions` with a callback argument named `send` or `trigger` (or something similar) makes the task comprehensible at a glance and greatly contributes in making an already highly nuanced framework more accessible to new users.
